src/Administration/Resources/app/administration/src/module/sw-login/view/sw-login-login/sw-login-login.html.twig

```diff
@@ -22,62 +22,79 @@
+                required
+            />
+                {% endblock %}
 
-    <sw-checkbox-field
-        v-model:value="rememberMe"
-        :label="$tc('sw-login.index.labelKeepLoggedIn')"
-    />
+            <sw-checkbox-field
+                v-model:value="rememberMe"
+                :label="$tc('sw-login.index.labelKeepLoggedIn')"
+            />
 
-    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-    {% block sw_login_login_submit %}
-    <div class="sw-login__submit">
-        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-        {% block sw_login_login_forgot_password %}
-        <router-link
-            :to="{ name: 'sw.login.index.recovery' }"
-            class="sw-login__forgot-password-action"
-        >
-            {{ $tc('sw-login.index.forgottenPasswordLink') }}
-        </router-link>
-        {% endblock %}
+            <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                {% block sw_login_login_submit %}
+            <div class="sw-login__submit">
+                <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                        {% block sw_login_login_forgot_password %}
+                <router-link
+                    :to="{ name: 'sw.login.index.recovery' }"
+                    class="sw-login__forgot-password-action"
+                >
+                    {{ $tc('sw-login.index.forgottenPasswordLink') }}
+                </router-link>
+                        {% endblock %}
 
-        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-        {% block sw_login_login_submit_button %}
-        <sw-button
-            :disabled="password.length <= 0 || username.length <= 0 || showLoginAlert"
-            class="sw-login__login-action"
-            variant="primary"
-        >
-            {{ $tc('sw-login.index.buttonLogin') }}
-        </sw-button>
-        {% endblock %}
+                <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                        {% block sw_login_login_submit_button %}
+                <sw-button
+                    :disabled="password.length <= 0 || username.length <= 0 || showLoginAlert"
+                    class="sw-login__login-action"
+                    variant="primary"
+                >
+                    {{ $tc('sw-login.index.buttonLogin') }}
+                </sw-button>
+                        {% endblock %}
+            </div>
+                {% endblock %}
+        </template>
+
+        <template v-for="(loginType, index) in loginConfig.providers" :key="index">
+            <div>
+                <a
+                    class="sw-button"
+                    :class="loginType.class"
+                    :disabled="isLoading"
+                    :href="loginType.url"
+                >
+                    {{ $tc(loginType.snippet_key) }}&nbsp;<sw-icon :name="loginType.icon"/>
```

Comment:
Is `&nbsp;` used to add a space here?  Why not just insert a line break and put `sw-icon` on the next line (should have the same outcome)?